### ISSUE TITLE
Python interface not planning to attached collision object subframe

### ISSUE THREAD

**Issue by vinceHuyghe** [ros-planning/moveit#3222]
### Description

When trying to plan with an attached collision object’s subframe set as the end effector link the following error is thrown: 
ERROR: Link 'end_effector/tcp' not found in model 'fanuc'

### Environment
* ROS Distro: Noetic 
* OS Version: Ubuntu 20.04
* Binary build
* latest moveit version

### Steps to reproduce
I have created a minimal example in following [repo](https://github.com/vinceHuyghe/example) this has been tested with [moveit_resources](https://github.com/ros-planning/moveit_resources) fanuc_moveit_config demo.launch and reproduces the error.

### Actual behaviour
ERROR: Link 'end_effector/tcp' not found in model 'fanuc'

**Comment by rhaschke** [comment]
Your example works for me w/o any warnings or errors. Do you have the `ResolveConstraintFrames` planning adapter activated?

**Comment by rhaschke** [comment]
This behavior resulted from a check to avoid planning if all goal constraints are already satisfied. Unfortunately, this check doesn't have the sub frames resolved yet and thus throws this error. I filed a simple fix: #3228.
Meanwhile, you can simply avoid the problem by planning and executing separately - `go()` performs both steps internally and your `plan()` call was actually redundant:

```python
        th.move_group.set_pose_target(pose)
        success, plan = th.move_group.plan()[:2]
        if success:
            th.move_group.execute(plan)
```

**Comment by vinceHuyghe** [comment]
Thank you. I will close the issue.
